pkg/crypto/bls/bbs12381g2pub/bls.go

```diff
@@ -0,0 +1,227 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package bbs12381g2pub
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"hash"
+
+	bls12381 "github.com/kilic/bls12-381"
+	"github.com/phoreproject/bls"
+	"golang.org/x/crypto/blake2b"
+)
+
+// BlsG2Pub defines BBS+ signature scheme where public key is a point in the field of G2.
+type BlsG2Pub struct {
+}
+
+// NewBlsG2Pub creates a new BlsG2Pub.
+func NewBlsG2Pub() *BlsG2Pub {
```

Comment:
Do you plan other constructors in this package?

Reply:
Actually no. There are several `Parse...()` methods only. So yes - renaming to just `New()`.